### PR TITLE
Fix 7181 | [Bug] Cannot update ToolbarItem text and icon

### LINKED ISSUE #7181: [Bug] Cannot update ToolbarItem text and icon
### Description
Cannot edit ToolbarItem Text and Icon in Shell ContentPage 4.1
### Steps to Reproduce

1. Created Shell application with ContentPage
2.  On OnAppearing and clicked events of the ToolbarItem, change the Text property
3. Text property won't change

### Expected Behavior
Text should be changed
### Actual Behavior
Text and Icons are not changed
### Basic Information
XF 4.1 Shell, Content Page, ToolbarItem

- Version with issue:
- Last known good version:
- IDE:
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages:
- Affected Devices:

### PR DESCRIPTION
**Pull request by felipebaltazar**
### Description of Change ###

I've created `ShellToolbarTracker.OnToolbarItemPropertyChanged` to get changes from `ToolbarItem.PropertyChanged` and update view

### Issues Resolved ### 

- fixes #7181

### API Changes ###
 
 None

### Platforms Affected ### 
- Android

### Behavioral/Visual Changes ###
ToolbarItem Text and Icons will be updated, when a new value set, on a shell page.

### Before/After Screenshots ### 

- Before

<p align="center">
	<kbd>
		image
	</kbd>
</p>

- After 

<p align="center">
	<kbd>
		image
	</kbd>
</p>

### Testing Procedure ###
1. Created Shell application with ContentPage
2. Set OnClicked events of the ToolbarItem, to change the Text property
3. Toolbaritem Text will change

> Or just run the Issue7181 test from ControlGallery

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/Renderers/ShellToolbarTracker.cs [hunk 1]
```diff
@@ -454,6 +454,9 @@ protected virtual void UpdateToolbarItems(Toolbar toolbar, Page page)
 
 			foreach (var item in page.ToolbarItems)
 			{
+				item.PropertyChanged -= OnToolbarItemPropertyChanged;
+				item.PropertyChanged += OnToolbarItemPropertyChanged;
```
Comment (rmarinho) [inline Xamarin.Forms.Platform.Android/Renderers/ShellToolbarTracker.cs]: We need to unsubscribe on dispose

**Comment by felipebaltazar** [comment]
> @felipebaltazar also the test is failing on iOS.

@rmarinho I can't be able to test on the IOS device since I don't have a MAC... So I've restricted the UI Test to the Android platform. I think this makes sense because the Issue targets the Android platform.

**Comment by jsuarezruiz** [comment]
Tested on iOS. It works as expected (same as Android).
issue7181-ios

**Comment by rmarinho** [comment]
Failing test on android is not related.

**Comment by jsuarezruiz** [comment]
@felipebaltazar This looks good to me, but can you rebase master Changs to fix the conflicts?. Thnx!

**Comment by felipebaltazar** [comment]
> @felipebaltazar This looks good to me, but can you rebase master Changs to fix the conflicts?. Thnx!

Yeah! I'll do this today at night (UTC-3:00)

**Pull request merged**